mars/dataframe/arithmetic/add.py

```diff
@@ -68,23 +68,23 @@ def _set_inputs(self, inputs):
             self._lhs = self._inputs[0]
             self._rhs = self._inputs[1]
         else:
-            if isinstance(self._lhs, DATAFRAME_TYPE):
+            if isinstance(self._lhs, (DATAFRAME_TYPE, SERIES_TYPE)):
                 self._lhs = self._inputs[0]
             elif np.isscalar(self._lhs):
                 self._rhs = self._inputs[0]
 
 
 def add(df, other, axis='columns', level=None, fill_value=None):
-    if isinstance(other, DATAFRAME_TYPE) or np.isscalar(other):
+    if isinstance(other, (DATAFRAME_TYPE, SERIES_TYPE)) or np.isscalar(other):
         op = DataFrameAdd(axis=axis, level=level, fill_value=fill_value, lhs=df, rhs=other)
     else:
-        raise NotImplementedError('Only support add with dataframe or scalar!')
+        raise NotImplementedError('Only support add with dataframe, series or scalar!')
```

Comment:
Pandas itself only supports "scalar, sequence, Series, or DataFrame", the sequences will be initialized as a series and throws `ValueError` otherwise.

Will revise to keep the same behavior with pandas.

Reply:
Fixed. I have left a comment about the current limitation in `testAddWithPlainValueUnaligned` and mark that test case as `expectedFailure`.